report_presentation/Presenter.py

```diff
@@ -7,4 +16,274 @@ def __init__(self, report_root, initial_warnings, final_warnings, changes):
+        html = ""
+        num = 0
+        for i in recommendations:
+            num += 1
+            message = i.message
+            if '<' in message:
+                message = message.replace('<', '&lt;')
+            if '>' in message:
+                message = message.replace('>', '&gt;')
+            html += '''
+                                <tr>
+                                        <td  class="serif" >%s</td>
+                                        <td  class="serif" >%s</td>
+                                        <td class="serif" >%s</td>
+                                        <td class="serif" >%s</td>
+                                        <td class="serif" >%s</td>
+                                        <td class="serif" >%s</td>
+                                </tr>
+                                ''' % (num, i.recommendation_code, i.file_path, i.project_name, i.line, message)
+        return html
+
+    def get_unique_filepath_and_warning_code(self, warnings):
+        unique_file_path = []
+        unique_warning_code = []
+        for i in warnings:
+            if i.warning_code not in unique_warning_code:
+                unique_warning_code.append(i.warning_code)
+            if i.file_path not in unique_file_path:
+                unique_file_path.append(i.file_path)
+        return unique_warning_code, unique_file_path
+
+    def get_string_of_list(self, alist):
+        astring = "    "
+        for i in alist:
+            astring += i
+            astring += " ;   "
+        return astring
+
+    def create_html(self, dpct_root, destination_root, all_warnings, final_warnings, changes, unique_warning_code, unique_file_path):
+        diff_path = Path.joinpath(Path.cwd(), 'html_files')
+        if diff_path.is_dir() and os.listdir(diff_path):
+            diff_html.remove_diff_folder(diff_path)
+        diff_html.find_dpcpp(dpct_root, destination_root, diff_path)
+
+        with open('report.html', 'w') as report:
+            report.write(
+                Presenter.html_page(self, all_warnings, final_warnings, changes, unique_warning_code, unique_file_path,
+                                    diff_path))
+        with open('subgraphs.html', 'w') as sub_images:
+            sub_images.write(sub_graph.add_images())
+
+    def remove_image_folder(self, image_path):
+        for i in os.listdir(image_path):
+            os.remove(os.path.join(image_path, i))
+        os.rmdir(image_path)
+
+    def show_visualize(self, report_root, all_warnings):
+        image_path = Path.joinpath(report_root, 'images')
+        if image_path.is_dir() and os.listdir(image_path):
+            Presenter.remove_image_folder(self, image_path)
+        image_path.mkdir(parents=True, exist_ok=True)
+        graph.visualization_overall(all_warnings, image_path)
+        graph.visualization_partial(all_warnings, image_path)
+
+
+def get_warnings_and_changes(dpct_root, destination_root):
```

Comment:
this function and the one below are not part of the class. Why not? Typically, there's only one class in each file, because it is easy to miss that these "lose" functions are actually not part of the class. Maybe move them to a separate file, or integrate them into the class?